releasenotes/notes/oracle_compliation-c5d75e0aaf04c8f1.yaml

```diff
@@ -0,0 +1,24 @@
+---
+features:
+  - |
+    This first version of an oracle compiler allows to "compile" typed python functions (operating only on bits (Int1) ) into quantum circuits.
+
+    .. jupyter-execute:
+
+      from qiskit.transpiler.oracle_compiler import compile_oracle
+      from qiskit.transpiler.oracle_compiler.types import Int1
+
+      def grover_oracle(a: Int1, b: Int1, c: Int1, d: Int1) -> Int1:
+          return (not a and b and not c and d)
+
+      network = compile_oracle(grover_oracle)
+      quantum_circuit = network.synth()
+      quantum_circuit.draw()) 
+
+    The parameter ``arg_regs`` in :meth:`qiskit.transpiler.oracle_compiler.LogicNetwork.synth` creates a circuit with registers refering to the parameter names:
+
+    .. jupyter-execute:
+
+      quantum_circuit = network.synth(arg_regs=True)
+      quantum_circuit.draw()) 
+
```

Comment:
Can you add a comment here about tweedledum being an optional dependency for here. This is is something users will need to know for using the new feature.

Reply:
done in 52a044893